lsif/src/worker.ts

```diff
@@ -59,49 +56,67 @@ const STORAGE_ROOT = process.env.LSIF_STORAGE_ROOT || 'lsif-storage'
-    repository: string,
-    commit: string,
-    filename: string
-): Promise<void> => {
-    console.log(`Converting ${repository}@${commit}`)
-
-    const input = fs.createReadStream(filename)
-    const tempFile = path.join(STORAGE_ROOT, 'tmp', uuid.v4())
-
-    try {
-        // Create database in a temp path
-        const { packages, references } = await convertLsif(input, tempFile)
-
-        // Move the temp file where it can be found by the server
-        await fs.rename(tempFile, createDatabaseFilename(STORAGE_ROOT, repository, commit))
-
-        // Add the new database to the xrepo db
-        await xrepoDatabase.addPackagesAndReferences(repository, commit, packages, references)
-    } catch (e) {
-        // Don't leave busted artifacts
-        await fs.unlink(tempFile)
-        throw e
-    }
+function createConvertJob(
+    xrepoDatabase: XrepoDatabase,
+    logger: Logger
+): (repository: string, commit: string, filename: string) => Promise<void> {
+    return async (repository, commit, filename) => {
+        const jobLogger = logger.child({ jobId: uuid.v4(), repository, commit })
+        const jobTimer = jobLogger.startTimer()
+        jobLogger.info('converting LSIF data')
+
+        const input = fs.createReadStream(filename)
+        const tempFile = path.join(STORAGE_ROOT, 'tmp', uuid.v4())
+
+        try {
+            // Create database in a temp path
+            const { packages, references } = await convertLsif(input, tempFile, jobLogger)
+
+            // Move the temp file where it can be found by the server
+            await fs.rename(tempFile, createDatabaseFilename(STORAGE_ROOT, repository, commit))
+
+            // Add the new database to the xrepo db
+            const xrepoTimer = jobLogger.startTimer()
+            jobLogger.debug('populating cross-repo database')
+            await xrepoDatabase.addPackagesAndReferences(repository, commit, packages, references)
+            xrepoTimer.done({ message: 'populated cross-repo database', level: 'debug' })
+        } catch (e) {
+            jobLogger.error('failed to convert LSIF data', { error: e })
+            // Don't leave busted artifacts
+            await fs.unlink(tempFile)
+            throw e
+        }
+
+        jobTimer.done({ message: 'converted LSIF data', level: 'info' })
 
-    // Remove input
-    await fs.unlink(filename)
+        // Remove input
+        await fs.unlink(filename)
+    }
 }
 
 /**
  * Create a job that updates the commits and branch heads for all repositories known
  * to the LSIF database (anything that has had LSIF data uploaded).
  *
  * @param xrepoDatabase The cross-repo database.
+ * @param logger The worker's logger instance.
  */
-const createUpdateCommitJob = (xrepoDatabase: XrepoDatabase): (() => Promise<void>) => () =>
-    updateCommits(GITSERVER_URLS, xrepoDatabase)
+const createUpdateCommitJob = (xrepoDatabase: XrepoDatabase, logger: Logger): (() => Promise<void>) => async () => {
```

Comment:
```suggestion
const createUpdateCommitJob = (xrepoDatabase: XrepoDatabase, logger: Logger) => async (): Promise<void> => {
```

This avoids the repeated parenthesis, which are difficult to read because it is not clear which are types and which are values.